src/Layout.js

```diff
@@ -1122,8 +1122,50 @@ var computeLayout = (function() {
       child.nextAbsoluteChild = null;
     }
   }
+  
+  function layoutNode(node, parentMaxWidth, parentDirection) {
+    if (node.isDirty ||
+        !node.lastLayout || 
+         node.lastLayout.requestedHeight !== node.layout.height ||
+         node.lastLayout.requestedWidth !== node.layout.width ||
+         node.lastLayout.parentMaxWidth !== parentMaxWidth) {
+           
+      if (!node.lastLayout) {
+        node.lastLayout = {};
+      }
+      
+      node.lastLayout.requestedWidth = node.layout.width;
+      node.lastLayout.requestedHeight = node.layout.height;
+      node.lastLayout.parentMaxWidth = parentMaxWidth;
+      
+      // Reset child layouts
+      node.children.forEach(function(child) {
+        child.layout.width = undefined;
+        child.layout.height = undefined;
+        child.layout.top = 0;
+        child.layout.left = 0;
+        child.layout.bottom = 0;
+        child.layout.right = 0;
+      });
+      
+      layoutNodeImpl(node, parentMaxWidth, parentDirection);
+      
+      for (var key in node.layout) {
+        node.lastLayout[key] = node.layout[key];
+      }
+      
+      node.isDirty = false;
```

Comment:
This isn't this function job to dirty/clean the nodes. It's up to whoever processes the update (see shouldUpdate)